### DISCUSSION TITLE
Random transcoding port assignment on startup

### DISCUSSION THREAD

**Discussion by nibreh** [owncast/owncast#3929] [Q&A]
Hi. I’m a beginner and still learning, but I believe I’ve identified the root cause of my issue.

Every time I start Owncast, the FFmpeg transcoding process attempts to connect to a random port (e.g., 127.0.0.1:46795 or similar) . This makes it difficult to manage firewall settings and port configurations.

I would like to know if it's possible to configure Owncast to use a fixed port for transcoding, instead of generating a new one at each startup. Ports 8080 and 1935 work fine for the main server, but the transcoding port keeps changing, and I would prefer not to open a wide range of ports. (or maybe it's not a problem ?)

Thanks in advance for your help!

Juh

**Comment by gabek** [comment]
This is purely an internal process on the same machine, I'm curious why you would need to open up a firewall port since it's all local?

But there is no way to change this. This is a convenience, as it picks any available port, since the port itself doesn't matter. It's one less thing people have to worry about when it comes to multiple applications trying to use the same port or running multiple copies of Owncast on the same machine.

**Comment by nibreh** [comment]
Thank you for the clarification. I’m using a VPS where Owncast is installed, and I'm streaming with FFmpeg from the same server. I will do this as an experiment to learn and try.

I understand that I can add the following rule to my iptables to address the port issue:

`sudo iptables -A INPUT -p tcp -s 127.0.0.1 --dport 30000:50000 -j ACCEPT`

Since this rule only allows local connections, it should not pose any security risks ? 

I’m also aware that my iptables configuration might not be optimal ...

Thank you for your help!
